tests/plugin.test.js

```diff
@@ -464,3 +475,214 @@ test.serial(
+        collect: {
+            timing: false,
+            hits: false,
+            errors: false,
+            health: false,
+        },
+    });
+    t.true(server.hasDecorator('timerify'));
+    const timerified = server.timerify('func', func);
+    await Promise.all([
+        new Promise((resolve) => {
+            t.context.statsd.on('metric', (buffer) => {
+                t.true(/ns\.func:\d+(\.\d+)?\|ms/.test(buffer.toString()));
+                resolve();
+            });
+        }),
+        timerified(),
+    ]);
+    t.pass();
+});
+
+test.serial('timerify custom send implementation', async (t) => {
+    const onSend = sinon.spy();
+    const func = async () => {
+        await sleep(100);
+    };
+
+    const server = await setup({
+        host: `udp://127.0.0.1:${t.context.address.port}`,
+        namespace: 'ns',
+        collect: {
+            timing: false,
+            hits: false,
+            errors: false,
+            health: false,
+        },
+    });
+    const timerified = server.timerify('func', func, onSend);
+    await timerified();
+    // The call to the perf observer callbakc is not immediate, let's wait a bit.
+    await sleep(100);
+    t.true(onSend.calledOnce);
+    t.is(server, onSend.firstCall.thisValue);
+    t.is('func', onSend.firstCall.firstArg);
+    t.true(
+        typeof onSend.firstCall.lastArg === 'number' ||
+            typeof onSend.firstCall.lastArg === 'object'
+    );
+});
+
+test.serial('timerify custom onSend on Node >= 16', async (t) => {
+    // No way to conditionally skip a test
+    if (!gte16) {
+        t.log('skipping test');
+        return t.pass();
+    }
+    const { PerformanceEntry } = require('perf_hooks');
+
+    const onSend = sinon.spy();
+    const asyncFunc1 = async () => {
+        await sleep(100);
+    };
+    const asyncFunc2 = async () => {
+        await sleep(200);
+    };
+    const syncFunc = () => {};
+
+    const server = await setup({
+        host: `udp://127.0.0.1:${t.context.address.port}`,
+        namespace: 'ns',
+        collect: {
+            timing: false,
+            hits: false,
+            errors: false,
+            health: false,
+        },
+    });
+    const asyncTimerified1 = server.timerify('asyncFunc1', asyncFunc1, onSend);
+    const asyncTimerified2 = server.timerify('asyncFunc2', asyncFunc2, onSend);
+    const syncTimerified = server.timerify('syncFunc', syncFunc, onSend);
+    const syncArrowDecorator = server.timerify(
+        'syncArrowDecorator',
+        server.syncArrow
+    );
+    const syncAnonDecorator = server.timerify(
+        'syncAnonDecorator',
+        server.syncAnon
+    );
+    const syncDecorator = server.timerify('syncDecorator', server.syncFunc);
+    const anon = server.timerify('anon', () => {});
+    const asyncJobs = Promise.all([asyncTimerified2(), asyncTimerified1()]);
+
+    syncTimerified();
+    // All the anonymous functions will conflict with each other, resulting in onSend getting
```

Comment:
This needs to documented

Reply:
Yea, as we agreed in prev call, we should document this, there is no way to avoid clashing when arrow functions are used directly as parameters. It's a gotcha that will probably generate confusion but I see no other way ATM.